spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,185 @@
+            basis_type,
+            normalization,
+            channel_convention,
+            phase_convention,
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """Create SphericalHarmonicSignal with data, and sampling rate.
+
+        Parameters
+        ----------
+        data : ndarray, double
+            Raw data of the spherical harmonics signal in the time or
+            frequency domain. The memory layout of data is 'C'. E.g.
+            data of ``shape = (3, 4, 1024)`` has 3 channels with 4
+            spherical harmonics coefficients with 1024 samples or frequency
+            bins each. Time data is converted to ``float``. Frequency is
+            converted to ``complex`` and must be provided as single
+            sided spectra, i.e., for all frequencies between 0 Hz and
+            half the sampling rate.
+        sampling_rate : double
+            Sampling rate in Hz
+        n_max : int
+            Maximum spherical harmonic order. Has to match the number of
+            coefficients, such that the number of coefficients
+            >= (n_max + 1) ** 2.
+        basis_type : str
+            Type of spherical harmonic basis, either ``'complex'`` or
+            ``'real'``. The default is ``'complex'``.
+        normalization : str
+            Normalization convention, either ``'n3d'``, ``'maxN'`` or
+            ``'sn3d'``. The default is ``'n3d'``.
+            (maxN is only supported up to 3rd order)
+        channel_convention : str
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        phase_convention : str
+            Either 'condon-shortley', or 'ambix'.
+            TODO: discuss ...
+        n_samples : int, optional
+            Number of samples of the time signal. Required if domain is
+            ``'freq'``. The default is ``None``, which assumes an even number
+            of samples if the data is provided in the frequency domain.
+        domain : ``'time'``, ``'freq'``, optional
+            Domain of data. The default is ``'time'``
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+            or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+            for more information. The default is ``'none'``, which is typically
+            used for energy signals, such as impulse responses.
+        comment : str
+            A comment related to `data`. The default is ``None``.
+
+        References
+        ----------
+        ..
+
+        """
+
+        if not data.shape[-2] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few sh coefficients '
+                             'for n_max = {n_max}.')
```

Comment:
Should be a single line f-string or multiple line f-string in paranthesis to avoid line breaks in the error message. Did you know this trick, which prints exactly what you want:
```suggestion
            raise ValueError(f'Data has to few sh coefficients for {n_max =}.')
```
Maybe add the highest possible n_max to the message to help the user resolve the error?